examples/cluster/plot_digits_linkage.py

```diff
@@ -50,13 +43,13 @@ def plot_clustering(X_red, labels, title=None):
     X_red = (X_red - x_min) / (x_max - x_min)
 
     plt.figure(figsize=(6, 4))
-    for i in range(X_red.shape[0]):
-        plt.text(
-            X_red[i, 0],
-            X_red[i, 1],
-            str(y[i]),
-            color=plt.cm.nipy_spectral(labels[i] / 10.0),
-            fontdict={"weight": "bold", "size": 9},
+    for t in np.sort(np.unique(y)):
```

Comment:
Instead of make the sorting we can load earlier the full Bunch:

```
digits = datasets.load_digits()
X, y = digits.data, digits.target
```

and then use the `target_names` that will be the array that you are expecting with the unique + sort.

```suggestion
    for digit in digits.target_names:
```